rust/private/rust.bzl

```diff
@@ -1115,6 +1124,7 @@ rust_test = rule(
         str(Label("//rust:toolchain_type")),
         "@bazel_tools//tools/cpp:toolchain_type",
     ],
+    cfg = force_panic_unwind_transition,
```

Comment:
Can this not also be solved by having this flag appear in the `rust_toolchain` instead of for each test and proc-macro? Also, why is this not for binaries as well?

Reply:
It needs to apply only to tests (the overall goal of this PR is build binaries without this transition). `#[should_panic]` doesn't work with `-Cpanic=abort`, so Cargo automatically builds tests with `-Cpanic=unwind`, which is the behavior I'm copying here.

https://doc.rust-lang.org/cargo/reference/profiles.html#panic documents this:
> Tests, benchmarks, build scripts, and proc macros ignore the `panic` setting. The `rustc` test harness currently requires `unwind` behavior. See the [panic-abort-tests](https://doc.rust-lang.org/cargo/reference/unstable.html#panic-abort-tests) unstable flag which enables `abort` behavior.
>
> Additionally, when using the `abort` strategy and building a test, all of the dependencies will also be forced to build with the `unwind` strategy.